### PR TITLE
allow registering custom validators

### PR DESCRIPTION
**Pull request by Rahien**
image
image
image

### CODE REVIEW

File: src/constraints.js [hunk 1]
```diff
@@ -26,59 +26,121 @@ import hasOneNumberGreaterThanInFields from "./constraints/has-one-number-greate
...
-    case "http://lblod.data.gift/vocabularies/forms/HasOneNumberGreaterThanInFields":
-      return hasOneNumberGreaterThanInFields;
-    case "http://lblod.data.gift/vocabularies/forms/MatchValues":
-      return matchValues;
-    case "http://lblod.data.gift/vocabularies/forms/DateInPast":
-      return dateInPast;
-    default:
-      return false; //TODO: TBD
-  }
+  // TBD: what happens if the constraint isn't found, currently return false, wouldn't it be better to throw an error or return null?
```
Comment (Rahien) [inline src/constraints.js:142]: Left in the currenctly existing TBD comment and explained what i think it means

File: src/constraints.js [hunk 2]
```diff
@@ -26,59 +26,121 @@ import hasOneNumberGreaterThanInFields from "./constraints/has-one-number-greate
...
+  return CUSTOM_VALIDATIONS.get(uri) || BUILT_IN_VALIDATIONS.get(uri) || false;
```
Comment (Rahien) [inline src/constraints.js:143]: this order is important. it allows you to override a built in validation if you so wish

File: src/constraints.js [hunk 3]
```diff
@@ -26,59 +26,121 @@ import hasOneNumberGreaterThanInFields from "./constraints/has-one-number-greate
...
+  );
+}
+
+registerDefaultValidations();
+
+export function registerCustomValidation(uri, validator) {
+  CUSTOM_VALIDATIONS.set(uri, validator);
+}
+
+export function resetCustomValidations() {
```
Comment (Rahien) [inline src/constraints.js:137]: not sure if actually useful, but it existed for the custom components that i based myself on, so kept it in.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,17 +1,20 @@
 ## v2.9.0 (2024-01-29)
 
 #### :rocket: Enhancement
-* [#63](https://github.com/lblod/submission-form-helpers/pull/63) [DL-5623] Add a new `DateInPast` constraint ([@Windvis](https://github.com/Windvis))
+
```
[the remark this answers is not in the document]
Comment (Rahien) [reply CHANGELOG.md:4]: :thought_balloon: might be nice to either ignore or fix for new things?

**Review by Windvis — APPROVED** [review]
Nice, thanks for working on this!

**Comment by Windvis** [comment]
Released as [v2.10.0](https://github.com/lblod/submission-form-helpers/blob/master/CHANGELOG.md#v2100-2024-03-26)

**Pull request merged**